library/L1_Peripheral/lpc40xx/eeprom.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "L1_Peripheral/eeprom.hpp"
+
+#include "L0_Platform/lpc40xx/LPC40xx.h"
+#include "L1_Peripheral/lpc40xx/system_controller.hpp"
+
+namespace sjsu
+{
+namespace lpc40xx
+{
+class Eeprom final : public sjsu::Eeprom
+{
+ public:
+  inline static LPC_EEPROM_TypeDef * eeprom_register = LPC_EEPROM;
+
+  static constexpr uint8_t kProgramStatusBit   = 28;
+  static constexpr uint8_t kReadWriteStatusBit = 26;
+
+  // EEPROM Command codes
+  static constexpr uint8_t kRead32Bits    = 0b010;
+  static constexpr uint8_t kWrite32Bits   = 0b101;
+  static constexpr uint8_t kEraseProgram  = 0b110;
+
+  // Getting actual SystemController Object
+  static constexpr sjsu::lpc40xx::SystemController kLpc40xxSystemController =
+      sjsu::lpc40xx::SystemController();
+
+  // Constructor has an optional SystemController parameter which can be
+  // specified for testing purposes. Otherwise, the actual SystemController
+  // object is used (kLpc40xxSystemController)
+  explicit constexpr Eeprom(const sjsu::SystemController & system_controller =
+                             kLpc40xxSystemController)
+      : system_controller_(system_controller)
+  {
+  }
+
+  // Initializing the EEPROM requires setting the wait state register,
+  // and the clock divider register
+  void Initialize() const override

```

Comment:
Does the EEPROM need to be powered on?

Reply:
The EEPROM is powered on by default on startup, but it can be powered down to save power.

-Table 731 from UM10562.pdf

Reply:
Hmmm, run the power on anyway, in the event someone has powered it down for some reason, but then wants to power it back up later.